lib/internal/test_runner/mock/mock_timers.js

```diff
@@ -98,7 +101,75 @@ class MockTimers {
     this.#executionQueue.removeAt(position);
   }
 
-  async * #setIntervalPromisified(interval, startTime, options) {
+  #createDate() {
+    kMock = Symbol('MockTimers');
+    /**
+     * Function to mock the Date constructor, treats cases as per ECMA-262
+     * and returns a Date object with a mocked implementation
+     * @typedef {Date} MockDate
+     * @returns {MockDate} a mocked Date object
+     */
+    function MockDate(year, month, date, hours, minutes, seconds, ms) {
```

Comment:
not sure if jsdocs works inside a private function

Reply:
Those are just intended to be internal documentation, it's not intended to be shown